### DISCUSSION TITLE
Simple Tensor identities

### DISCUSSION THREAD

**Discussion by wedelfach** [FeynCalc/feyncalc#82] [Q&A] [answered]
Hi,

I am using FeynCalc 9.3.1 in which version there is an implementation of a tensor called TensorFunction. I would like to ask wheter it is possible to teach Mathematica to perfom easy simplifications of TensorFunction expressions given some rules eg.:
T[A,i,j]T[A,j,k]=C_2[i,k]
T[A,i,j]Y[s,j,k]-Y[s,i,j]T[A,j,k]=U[A,s,p]Y[p,i,k]
T[A,i,j]T[A,j,i]=S_2(F)(A,B)
where I have used Einstein summation convention.
In other worlds I'm trying to implement "Reduce" for TensorFunctions. Please let me know if you have any ideas how it can be done.

**Comment by vsht** [comment, marked as answer]
I think one could do something along these lines, but it won't feel very natural.
In particular, to perform the simplifications you mention, one would have to collect
w.r.t indices and then apply suitable replacement rules to the isolated products of tensors.

Most likely the performance will be not so good, unless you plan to look only at very
short expressions.

**Reply by wedelfach** [reply]
Thank you! So, if I understood you correctly, FeynCalc is not the best tool to simplify long tensor expressions and I should look for some other package.

**Reply by vsht** [reply]
Well, in a typical particle physics calculations, the tensors you usually have are 4-vectors, the metric tensor and the Levi-Civita tensor.
In addition to that there are Dirac matrices with Lorentz indices attached to them. For that kind of stuff FeynCalc works sufficiently well
(at least as long as the number of terms doesn't become too large for Mathematica to handle). 

If there are few more custom tensors appearing in the calculation, one can always cook up something for them. But if the calculation
consists of manipulating various nontrivial tensors with special symmetries, then FeynCalc is probably not the right tool. I suppose that in this case you should look in the direction of xAct/Cadabra/RedBerry etc.
